src/Logic/CoreRules.ml

```diff
@@ -1243,8 +1268,8 @@ let rfind (bd,body) arg asym fvs ju =
       tacerror "find not a closed function: %a contains variables not in %a"
         pp_exp body (pp_list "," pp_exp) (Se.elements allowed);
     let e1 = e_subst subst_bd body in
-    if not (e_equal e1 ev.ev_expr) then
```

Comment:
This should not be in a core rule:
There can be a high-level rule that applies norm first (to the event).